### PR TITLE
Added table based auto complete select widget fixes #1622

### LINKED ISSUE #1622: Auto Complete Widget
Add a widget that will look at at table of unique values and as a user types fetch matching results.

### PR DESCRIPTION
**Pull request by mounicarachakonda**

### CODE REVIEW

**Comment by jbrundage** [comment]
Hi Mounica, Jacob asked me to review your pull request. I've glanced over the files and generally things are looking good. I'll fully review it by the end of the day today so that it will be merge-ready by the time Gordon sees it.

**Comment by jbrundage** [comment]
@mounicarachakonda Include this auto-complete javascript file via adding it as a dependency in our bower.json file. "javascript-auto-complete"

File: test/otherFactory.js [hunk 1]
```diff
@@ -87,23 +111,17 @@
...
-                            ["Geography II", 16, 73, 52, 83],
-                            ["Width 2 undefined", , 83, , 72],
-                            ["Science", 66, 60, 85, 6],
-                            ["Science II", 46, 20, 53, 7],
-                            ["With 2 NULL", null, 20, null, 7],
-                            ["Math", 98, 30, 23, 13],
-                            ["Math II", 76, 30, 34, 6],
-                            ["Math III", 80, 30, 27, 8]
-                        ])
+                        .columns(["aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "baaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "caaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"])
```
Comment (jpellock) [inline test/otherFactory.js]: did you mean to leave this in?

Comment (mounicarachakonda) [inline test/otherFactory.js]: This was there previously. I have not added it.

**Comment by GordonSmith** [comment]
@mounicarachakonda This is failing a lint test (if you click on the "details" link next to the red x, it should show you the info.

Alternatively you can call ```gulp lint``` locally to test.

**Comment by GordonSmith** [comment]
Closing in favor of:  https://github.com/hpcc-systems/Visualization/pull/1833#partial-pull-merging

**Pull request closed without merging**